### PR TITLE
BugFix:  Some error results have messages that are not reporting

### PR DESCRIPTION
**Pull request by matthewvon**
Discovered that some rocksdb error messages were not getting copied all the way to the client.  The generic error number arrived, but not the specific error message.  This updates code at the coordinator level to always report error messages retrieved from the agency updates.

- [XX] Bug-Fix for at least devel and 3.4 ... considering 3.3
- [XX ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] The behaviour change can only be verified via automatic tests

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by matthewvon** [comment]
Running:  http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5169/

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -3038,7 +3038,7 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
                   "rolling back index creation.");
             }
 
-            return Result(tmpRes);
+            return Result(tmpRes, *errMsg);
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp]: A [few lines above](https://github.com/arangodb/arangodb/pull/9454/files#diff-ab0fffbe5ac5ed0a2189cc76962b080dR3002) there is a comment saying
> // The mutex in the condition variable protects the access to
> // *errMsg:

I think we should use the CONDITION_LOCKER here too.

File: arangod/Cluster/ClusterInfo.cpp [hunk 2]
```diff
@@ -3057,7 +3057,7 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
                   "Timed out while trying to roll back index creation failure");
             }
 
-            return Result(tmpRes);
+            return Result(tmpRes, *errMsg);
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp]: same here

**Comment by matthewvon** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5177/

**Pull request merged**